purchase_add_products_from_bom/wizard/add_products_from_bom.py

```diff
@@ -0,0 +1,387 @@
+            ])
+
+            if product_already_in_line:
+                product_already_in_line.product_qty += line.product_qty
+            else:
+                values = self._get_values_purchase_line(line)
+                self.purchase_id.order_line.create(values)
+
+    def _get_values_purchase_line(self, line):
+        values = self._simulate_purchase_line_onchanges(
+            line.product_id,
+            line.product_qty,
+            line.product_uom_id,
+        )
+        values['order_id'] = self.purchase_id.id
+        # Price informed by user has priority
+        values['price_unit'] = line.price_unit
+        return values
+
+
+class AddProductsFromBoMLines(models.TransientModel):
+    _name = "add.products.from.bom.lines"
+    _description = "Purchase Add Products From BoM Line"
+
+    # Use in view to show or not Button to explode semi finished products
+    has_bom = fields.Boolean(compute='_compute_has_bom')
+
+    product_finished_id = fields.Many2one(
+        comodel_name='add.products.from.bom'
+    )
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product'
+    )
+    orig_bom_product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Origin BoM Product',
+        readonly=True,
+    )
+    bom_id = fields.Many2one(
+        comodel_name='mrp.bom',
+        string='Bill of Material',
+
+    )
+    # Used to create Domain
+    bom_ids = fields.One2many(
+        comodel_name='mrp.bom',
+        related='product_id.product_tmpl_id.bom_ids',
+        string='Bill of Materials'
+    )
+    product_qty = fields.Float(
+        string='Quantity to Buy',
+        digits=dp.get_precision('Product Unit of Measure'),
+        required=True
+    )
+    product_uom_id = fields.Many2one(
+        comodel_name='uom.uom',
+        string='Unit of Measure'
+    )
+    purchase_uom_id = fields.Many2one(
+        comodel_name='uom.uom',
+        string='Purchase Unit of Measure'
+    )
+    # fields.Monetary required Currency field to avoid error:
+    #  AssertionError: Field purchase.add.products.from.bom.line.price_unit
+    #  with unknown currency_field None
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+        string='Currency',
+        related='product_finished_id.currency_id',
+    )
+    price_unit = fields.Monetary(
+        string='Unit Price',
+        currency_field='currency_id',
+    )
+    product_qty_stock = fields.Float(
+        string='Quantity Available in Stock',
+        related='product_id.qty_available'
+    )
+    product_qty_after_buy = fields.Float(
+        string='Quantity After Buy',
+        compute='_compute_product_qty_after_buy',
+    )
+    total_product_value = fields.Monetary(
+        string='Total Product Value',
+        currency_field='currency_id',
+        compute='_compute_total_product_value',
+    )
+    selected_product = fields.Boolean(string='Selected Product')
```

Comment:
I think it should be True by default. It was confusing for me : if i selected a product (and its bom), it is because i want it to be import in the purchase order :) And it's not perfectly clear that we have to select the lines for that !

Reply:
It was made in this way beacuse some products can has enough quantity in stock and don't need to buy more, that the reason to show in wizard the "Quantity Avaliable in Stock" and "Quantity after Buy" so the users should be able to decide based in those quantitys which products they want to import. But I don't see any problem in mark all lines with True and the users uncheck the lines

Reply:
Yes i understand. I think that it's quicker for a lot of use cases, to have all lines to True and uncheck them :)